undofx/src/main/java/org/fxmisc/undo/impl/UndoManagerImpl.java

```diff
@@ -209,7 +229,7 @@ private void addChange(C change) {
     @SuppressWarnings("unchecked")
     private C[] merge(C c1, C c2) {
         Optional<C> merged = merge.apply(c1, c2);
-        if(merged.isPresent()) {
+        if(merged.isPresent() && isNotIdentity.test(merged.get())) {
             return (C[]) new Object[] { merged.get() };
         } else {
             return (C[]) new Object[] { c1, c2 };
```

Comment:
> Now that you inlined it, I see one problem, though. If the changes can annihilate, it can cause merging of changes that were previously prevented from being merged.

To rephrase this in my own words...
Given a change that is prevented from being merged with another change (unmergeable change), a change annihilation that occurs directly after the unmergeable change should not allow the unmergeable change to be merged with another pushed change. This implies that `canMerge` will need to be set differently in different circumstances.